File: uncommon-words-from-two-sentences/uncommon-words-from-two-sentences.java
class Solution {
    public String[] uncommonFromSentences(String A, String B) {
        HashMap<String, Integer> count = new HashMap<String, Integer>();
        for(String s: (A + " " + B).split(" "))
            count.put(s, count.getOrDefault(s,0)+1);
        
        List<String> result = new LinkedList();
        for(String word: count.keySet()){
            if(count.get(word)==1){
                result.add(word);
            }
        }
        
        return result.toArray(new String[result.size()]);
    }
}

File: uncommon-words-from-two-sentences/uncommon-words-from-two-sentences.cpp
class Solution {
public:
    vector<string> uncommonFromSentences(string A, string B) {
        map<string, int> word_count;
        
        string temp = "";
        for(int i=0;i<A.size();i++){
            if(A[i]==' '){
                word_count[temp]++;
                temp = "";
                i++;
            }
            if(i==A.size()-1){
                temp += A[i];
                word_count[temp]++;
            }
            temp += A[i];
        }
        
        temp = "";
        for(int i=0;i<B.size();i++){
            if(B[i]==' '){
                word_count[temp]++;
                temp = "";
                i++;
            }
            if(i==B.size()-1){
                temp += B[i];
                word_count[temp]++;
            }
            temp += B[i];
        }
        
        vector<string> result;
        for(auto itr=word_count.begin();itr!=word_count.end();++itr){
            if(itr->second==1)
                result.push_back(itr->first);
        }
        
        return result;
    }
};